### PR TITLE
Update client support matrices

### PR DESCRIPTION
**Pull request by paususe**
Detailed list of changes and reviewers for confirmation:
- Package locking and system locking are now supported on Salt minions (SLES, SLES ES, CentOS, RHEL, Ubuntu, Debian). @mbologna @mateiw could you please confirm?
- Pre-downloading of packages now supported on Ubuntu and Debian. @admd @brejoc could you please confirm?
- Add SLES ES 8 to the SLES ES matrix, for some reason we only had SLES ES 6 and 7 but 8 was missing
- Remove admonition about Debian on the Ubuntu page when generating Uyuni docs. It does no make sense, since Uyuni has its own Debian page.
- Do not mention CentOS Expanded Support has a different level of support. That's SUSE Manager, on Uyuni, vanilla CentOS and CentOS + ES have the same level of support (community).

### CODE REVIEW

**Comment by mbologna** [comment]
>     * Package locking and system locking are now supported on Salt minions (SLES, SLES ES, CentOS, RHEL, Ubuntu, Debian). @mbologna @mateiw could you please confirm?

The behavior, the process, and the features allowed are totally
different for locked traditional and Salt systems.
Do we define in the documentation what is a locked system and what are the actions that are allowed while a system is locked?

File: modules/client-configuration/pages/supported-features-debian.adoc [hunk 1]
```diff
@@ -46,13 +46,13 @@ The icons in this table indicate:
...
 | Software crash reporting | icon:times[role="danger"] | icon:times[role="danger"]
-| Staging (pre-download of packages) |  icon:question[role="gray"] | icon:question[role="gray"]
+| Staging (pre-download of packages) |  ``Salt`` | ``Salt``
 | Duplicate package reporting | ``Salt`` | ``Salt``
 | CVE auditing |  icon:question[role="gray"] | icon:question[role="gray"]
 | SCAP auditing | icon:question[role="gray"] | icon:question[role="gray"]
 | Package verification | icon:times[role="danger"] | icon:times[role="danger"]
-| Package locking | icon:times[role="danger"] | icon:times[role="danger"]
-| System locking |  icon:times[role="danger"] | icon:times[role="danger"]
+| Package locking | ``Salt`` | ``Salt``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-debian.adoc:54]: AFAIK Package locking is not implemented for salt.

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 2]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
+| Remote commands | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| System package states | ``Salt`` | ``Salt`` | ``Salt``
+| System custom states | ``Salt`` | ``Salt`` | ``Salt``
+| Group custom states | ``Salt`` | ``Salt`` | ``Salt``
+| Organization custom states | ``Salt`` | ``Salt`` | ``Salt``
+| System set manager (SSM) | ``Salt`` | ``Salt`` | ``Salt``
+| Service pack migration | icon:times[role="danger"] | icon:times[role="danger"] | icon:times[role="danger"]
+| Basic Virtual Guest Management icon:asterisk[role="none"] | ``Traditional`` | icon:check[role="green"] | icon:check[role="green"]
+| Advanced Virtual Guest Management icon:asterisk[role="none"] | icon:times[role="danger"] | ``Salt`` | ``Salt``
+| Virtual Guest Installation (Kickstart), as Host OS | ``Traditional`` | ``Traditional`` | ``Traditional``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:34]: v8 does not have the "Traditional" Stack. So it is not supported

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 3]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
+| Virtual Guest Installation (image template), as Host OS | ``Traditional`` | icon:check[role="green"] | icon:check[role="green"]
+| System deployment (PXE/Kickstart) | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| System redeployment (Kickstart) | ``Traditional`` | icon:check[role="green"] | icon:check[role="green"]
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:37]: v8: Not tested and we know it is not working. Somebody need to invest some time to fix it before we publish this.

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 4]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
 | Contact methods | ``Traditional``: OSAD, RHNSD, SSH-push. +
 ``Salt``: ZeroMQ, Salt-SSH | ``Traditional``: OSAD, RHNSD, SSH-push. +
+``Salt``: ZeroMQ, Salt-SSH | ``Traditional``: OSAD, RHNSD, SSH-push. +
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:40]: v8: does not have "Traditional" Stack. No: ``Traditional``: OSAD, RHNSD

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 5]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
-| CVE auditing | icon:check[role="green"] | icon:check[role="green"]
-| SCAP auditing | icon:check[role="green"] | icon:check[role="green"]
-| Package verification | ``Traditional`` | ``Traditional``
-| Package locking | ``Traditional`` | ``Traditional``
-| System locking | ``Traditional`` | ``Traditional``
-| System snapshot | ``Traditional`` | ``Traditional``
-| Configuration file management | icon:check[role="green"] | icon:check[role="green"]
+| Works with {productname} Proxy | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| Action chains | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| Software crash reporting | icon:times[role="danger"] | ``Traditional`` | ``Traditional``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:44]: Not supported with v8 as we do not have the traditional stack.

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 6]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
+| Staging (pre-download of packages) | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| Duplicate package reporting | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| CVE auditing | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| SCAP auditing | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| Package verification | ``Traditional`` | ``Traditional`` | ``Traditional``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:49]: Not supported with v8

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 7]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
+| Package locking | ``Traditional`` | ``Traditional`` | icon:check[role="green"]
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:50]: Not on v8 as no traditional stack available.

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 8]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
+| System locking | ``Traditional`` | ``Traditional`` | icon:check[role="green"]
+| System snapshot | ``Traditional`` | ``Traditional`` | ``Traditional``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:52]: Not available for v8

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 9]
```diff
@@ -13,52 +13,54 @@ The icons in this table indicate:
...
+| Configuration file management | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
 | Snapshots and profiles |  ``Traditional``. +
 ``Salt``: Profiles supported, Sync not supported | ``Traditional``. +
+``Salt``: Profiles supported, Sync not supported | ``Traditional``. +
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:56]: Remove traditional stack and RHNSD, OSAD from the list of v8

File: modules/client-configuration/pages/supported-features-rh.adoc [hunk 10]
```diff
@@ -44,8 +44,8 @@ The icons in this table indicate:
 | CVE auditing | icon:check[role="green"] | icon:check[role="green"] | ``Salt``
 | SCAP auditing | icon:check[role="green"] | icon:check[role="green"] | ``Salt``
 | Package verification | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
-| Package locking | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
-| System locking | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
+| Package locking | ``Traditional`` | ``Traditional`` | icon:check[role="green"]
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-rh.adoc:47]: Not for v8

File: modules/client-configuration/pages/supported-features-sles.adoc [hunk 11]
```diff
@@ -49,8 +49,8 @@ The icons in this table indicate:
 | CVE auditing |  icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"] | ``Salt``
 | SCAP auditing |   icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"] | ``Salt``
 | Package verification | ``Traditional`` | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
-| Package locking |  ``Traditional`` | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
-| System locking |  ``Traditional`` | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
+| Package locking |  ``Traditional`` | ``Traditional`` | ``Traditional`` | icon:check[role="green"]
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-sles.adoc:52]: Only for traditional stack available. I am unsure if we ship it for openSUSE 15.1. I think not. So it is not available.

File: modules/client-configuration/pages/supported-features-ubuntu.adoc [hunk 12]
```diff
@@ -46,13 +46,13 @@ The icons in this table indicate:
...
 | Software crash reporting | icon:times[role="danger"] | icon:times[role="danger"]
-| Staging (pre-download of packages) |  icon:question[role="gray"] | icon:question[role="gray"]
+| Staging (pre-download of packages) |  ``Salt`` | ``Salt``
 | Duplicate package reporting | ``Salt`` | ``Salt``
 | CVE auditing |  icon:question[role="gray"] | icon:question[role="gray"]
 | SCAP auditing | icon:question[role="gray"] | icon:question[role="gray"]
 | Package verification | icon:times[role="danger"] | icon:times[role="danger"]
-| Package locking | icon:times[role="danger"] | icon:times[role="danger"]
-| System locking |  icon:times[role="danger"] | icon:times[role="danger"]
+| Package locking |  ``Salt`` | ``Salt``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-ubuntu.adoc:54]: No Package locking for salt

**Comment by Loquacity** [comment]
https://github.com/SUSE/spacewalk/issues/10773

File: modules/client-configuration/pages/supported-features-centos.adoc [hunk 13]
```diff
@@ -52,8 +52,8 @@ The icons in this table indicate:
 | CVE auditing (requires CVE ID) |  icon:check[role="green"] | icon:check[role="green"] | ``Salt``
 | SCAP auditing | icon:check[role="green"] | icon:check[role="green"] | ``Salt``
 | Package verification | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
-| Package locking | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
-| System locking |  ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
+| Package locking | ``Traditional`` | ``Traditional`` | icon:check[role="green"]
```
[the remark this answers is not in the document]
Comment (mbologna) [reply modules/client-configuration/pages/supported-features-centos.adoc:55]: When a Salt system is locked, one consequence is that package actions are not allowed from Uyuni/SUSE Manager. But this does not mean that there a package lock at the package manager level. I agree no mark it as N/A for Salt.

File: modules/client-configuration/pages/supported-features-debian.adoc (hunk 1 above)
Comment (mbologna) [reply modules/client-configuration/pages/supported-features-debian.adoc:54]: Agreed, see above.

**Comment by Loquacity** [comment]
@paususe what is the status on this?

**Comment by Loquacity** [comment]
Jumping into this because we need to get it in. Also partly addresses https://github.com/SUSE/spacewalk/issues/11361

File: modules/client-configuration/pages/supported-features-centos.adoc [hunk 14]
```diff
@@ -52,8 +52,8 @@ The icons in this table indicate:
...
+| System locking |  ``Traditional`` | ``Traditional`` | icon:check[role="green"]
```
[the remark this answers is not in the document]
Comment (Loquacity) [reply modules/client-configuration/pages/supported-features-centos.adoc:56]: I agree that we need to define this better, but I think I would prefer to do that as a separate PR. 
Issue: https://github.com/SUSE/spacewalk/issues/11374

[the remark this answers is not in the document]
Comment (mbologna) [reply modules/client-configuration/pages/supported-features-centos.adoc]: > I agree that we need to define this better, but I think I would prefer to do that as a separate PR.
> Issue: [SUSE/spacewalk#11374](https://github.com/SUSE/spacewalk/issues/11374)

In this case, I am OK to merge this PR, but I am also suggesting that we do not publish the manual until #11374 gets worked on. Approving.

File: modules/client-configuration/pages/supported-features-es.adoc [hunk 15]
```diff
@@ -13,52 +13,52 @@ The icons in this table indicate:
...
+| Action chains | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| Software crash reporting | icon:times[role="danger"] | ``Traditional`` | icon:times[role="danger"]
+| Staging (pre-download of packages) | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| Duplicate package reporting | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| CVE auditing | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| SCAP auditing | icon:check[role="green"] | icon:check[role="green"] | icon:check[role="green"]
+| Package verification | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
+| Package locking | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
+| System locking | ``Traditional`` | ``Traditional`` | icon:times[role="danger"]
+| System snapshot | ``Traditional`` | ``Traditional`` | ``Traditional``
```
Comment (mcalmer) [inline modules/client-configuration/pages/supported-features-es.adoc:51]: The problem with ES8 is, that we only support "salt". There is no traditional stack. So saying `traditional` here is wrong.

We should in general think about what we do with these OSes which only support `salt`.
As documented [check] mean both. That means all `[check]` in ES8 needs to be changed to `salt`.

**Pull request merged**